perl-package/AI-MXNetCAPI/mxnet.i

```diff
@@ -0,0 +1,1558 @@
+%module "AI::MXNetCAPI"
```

Comment:
In most of the cases the variable names are changed, to indicated for swig which typemap to use, please refer to another file in the same dir called typemaps.i that does real work and translates combinations of variable names and types to a specific perl binding.
This had to been done on a function by function basis (the same as you guys had hardcoded python ctypes usage in your python interface that because you know how you want to use a particular variable).
There's no auto-generation possible for new or changed APIs (besides, changing the contract on swig level will break high level perl level), I see no alternative to this.